foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_11_0/HiveUniversalConfigYAMLMigration.java

```diff
@@ -0,0 +1,110 @@
+import org.yaml.snakeyaml.error.YAMLException;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+
+public class HiveUniversalConfigYAMLMigration extends AbstractAissembleMigration {
+    private static final Logger logger = LoggerFactory.getLogger(HiveUniversalConfigYAMLMigration.class);
+    private static final List<String> propertiesForConfigStore = Arrays.asList("javax.jdo.option.ConnectionUserName");
+    private static final List<String> mysqlAuthForConfigStore = Arrays.asList("username");
+    private static final HashSet<String> linesToRemove = new HashSet<>(Arrays.asList("username: hive", "- name: javax.jdo.option.ConnectionUserName" ,"value: hive", "description: Username to use against metastore database", "# NB: UCS"));
+    private static final String DEFAULT_HIVE_USERNAME_AND_PASSWORD = "hive";
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        if (file.getName().endsWith("values.yaml")) {
+            try {
+                YamlUtils.YamlObject yaml = YamlUtils.loadYaml(file);
+                if(yaml.hasObject("aissemble-hive-metastore-service-chart"))
+                {
+                    yaml = yaml.getObject("aissemble-hive-metastore-service-chart");
+                }else{
+                    return false;
+                }
+
+                if (yaml.hasObject("mysql", "auth")) {
+                    YamlUtils.YamlObject mysqlAuth = yaml.getObject("mysql", "auth");
+                    for(String key : mysqlAuth.keySet())
+                    {
+                        if(mysqlAuthForConfigStore.contains(key) && !mysqlAuth.get(key).equals(DEFAULT_HIVE_USERNAME_AND_PASSWORD))
+                        {
+                            return false;
+                        }
+                    }
+
+                }
+
+                if (yaml.hasList("configMap", "metastoreServiceConfig", "properties")) {
+                    List<YamlUtils.YamlObject> properties = yaml.getListOfObjects("configMap", "metastoreServiceConfig", "properties");
+                    for(YamlUtils.YamlObject property : properties)
+                    {
+                        if(propertiesForConfigStore.contains(property.get("name")) && !property.get("value").equals(DEFAULT_HIVE_USERNAME_AND_PASSWORD))
+                        {
+                            return false;
+                        }
+                    }
+                }
+                return true;
+
+            } catch (YAMLException e) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Failed to parse YAML file, likely because it is a Helm template: {}", file.getPath(), e);
+                } else {
+                    logger.info("Failed to parse YAML file, likely because it is a Helm template: {}", file.getName());
```

Comment:
This log message is not accurate as this migration should only be running against `values.yaml`.

Reply:
updated log